cdap-common/src/main/java/io/cdap/cdap/common/utils/FileUtils.java

```diff
@@ -97,6 +98,21 @@ public static String getExtension(String name) {
     return idx >= 0 ? name.substring(idx + 1) : "";
   }
 
+  /**
+   * Returns the name of a given file without the extension, if one is present
+   * @param file the file
+   * @return the name of the file without the extension
+   */
+  public static String getNameWithoutExtension(File file) {
+    String fileName = file.getName();
+    int idx = fileName.lastIndexOf('.');
+
+    if (file.isDirectory() || idx < 0) {
```

Comment:
Please note in javadoc that extension is retained for directories

Reply:
Actually the implementation doesn't mirror the `FileUtils.getExtension`, which is not good. I updated the implementation.